moveit_ros/moveit_servo/include/moveit_servo/command_processor.hpp

```diff
@@ -0,0 +1,122 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/*      Title     : command_processor.hpp
+ *      Project   : moveit_servo
+ *      Created   : 04/06/2023
+ *      Author    : Brian O'Neil, Andy Zelenak, Blake Anderson, V Mohammed Ibrahim
+ */
+
+#pragma once
+
+#include <variant>
+#include <control_toolbox/pid.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <tf2_eigen/tf2_eigen.hpp>
+
+#include <moveit/kinematics_base/kinematics_base.h>
+
+#include <moveit_servo/datatypes.hpp>
+#include <moveit_servo/status_codes.hpp>
+#include <moveit_servo/utils.hpp>
+
+namespace moveit_servo
+{
+
+class CommandProcessor
+{
+public:
+  CommandProcessor(const moveit::core::JointModelGroup* joint_model_group, moveit::core::RobotStatePtr& current_state,
+                   servo::Params& servo_params, StatusCode& servo_status);
+
+  /**
+   * \brief Compute the change in joint position for the given joint jog command.
+   * @param command The joint jog command.
+   * @return The joint position change required (delta).
+   */
+  Eigen::VectorXd jointDeltaFromCommand(const JointJog& command);
+
+  /**
+   * \brief Compute the change in joint position for the given twist command.
+   * @param command The twist command.
+   * @return The joint position change required (delta).
+   */
+  Eigen::VectorXd jointDeltaFromCommand(const Twist& command);
+
+  /**
+   * \brief Compute the change in joint position for the given pose command.
+   * @param command The pose command.
+   * @return The joint position change required (delta).
+   */
+  Eigen::VectorXd jointDeltaFromCommand(const Pose& command);
+
+private:
+  /**
+   * \brief Set the IK solver that servo will use. If the robot does not have one, inverse jacobian will be used instead.
+   */
+  void setIKSolver();
+
+  /**
+   * \brief Get the current status of servo.
+   * @return The current status.
+   */
+  StatusCode getStatus();
+
+  /**
+   * \brief Get the message associated with the current servo status.
+   * @return The status message.
+   */
+  const std::string getStatusMessage();
+
+  /**
+   * \brief Computes the required change in joint angles for given cartesian change, using the robots IK solver.
+   * @param carteisan_position_delta The change in cartesian position.
+   * @return The required joing angle deltas.
+   */
+  Eigen::VectorXd detlaFromIkSolver(const Eigen::VectorXd& cartesian_position_delta);
+
+  // Variables
+  size_t num_joints_;
+  const moveit::core::JointModelGroup* joint_model_group_;
+  moveit::core::RobotStatePtr& robot_state_;
+  servo::Params& servo_params_;
+  StatusCode& servo_status_;
+
+  kinematics::KinematicsBaseConstPtr ik_solver_ = nullptr;
+
+  uint64_t controller_period_;
+  std::map<std::string, control_toolbox::Pid> controllers_;
```

Comment:
Check if these are used. I don't see why they would be.
```suggestion
```

Reply:
Removed these as well, since the last update moves away from PID based control.